src/arbitrum/epoch-based-managed-vaults/stratagies/Y2KFinanceStrategy.sol

```diff
@@ -1,7 +1,126 @@
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Y2KFinanceStrategy is BaseStrategy {
+
+    /// @notice Address of the Y2K Vault Factory
+    address vaultFactory;
+
+    /// @notice Array of vaults that were used
+    address[] vaults;
+    /// @notice Array of vault IDs that were used
+    uint256[] vaultIDs;
+
+    /********************************** Constructor **********************************/
+
+    constructor(address _assetVault, address _platform, address _manager)
+        BaseStrategy(_assetVault, _platform, _manager) {
+            vaultFactory = address(0x984E0EB8fB687aFa53fc8B33E12E04967560E092);
+        }
+
+    /********************************** View Functions **********************************/
+
+    function isActive() public view override returns (bool) {
+        if (isStrategiesActiveOverride) return false;
+        if (IERC20(assetVaultPrimaryAsset).balanceOf(address(this)) > 0) return true;
+
+        address[] memory _vaults = vaults;
+        uint256[] memory _vaultIDs = vaultIDs;
+        for (uint256 i = 0; i < vaults.length; i++) {
+            if (IY2KVault(_vaults[i]).balanceOf(address(this), _vaultIDs[i]) > 0) return true;
+        }
+
+        return false;
+    }
+
+    /********************************** Manager Functions **********************************/
+
+    /// @dev Executes the strategy - deposit into a Y2K Risk/Hedge Vault
+    /// @dev _configData:
+    /// @dev _index - index of the vault in the vaultFactory, determines the asset and the strike vault
+    /// @dev _amount - amount of the assetVaultPrimaryAsset to deposit, will be set to the balance if set to type(uint256).max
+    /// @dev _id - id of the vault in the vaultFactory, determines end date of the Epoch
+    /// @dev _type - true for Risk Vault, false for Hedge Vault
+    function execute(bytes memory _configData) external override onlyManager returns (uint256) {
+        
+        (uint256 _id, uint256 _amount, uint256 _before, address _vault) = _getConfig(_configData);
+
+        _approve(assetVaultPrimaryAsset, _vault, _amount);
+        IY2KVault(_vault).deposit(_id, _amount, address(this));
+
+        uint256 _shares = IY2KVault(_vault).balanceOf(address(this), _id) - _before;
+
+        // TODO
+        // if (_getStakingContract(_vault, _id) != address(0)) _stake(_shares);
+
+        return _shares;
+    }
+
+    /// @dev Terminates the strategy - withdraw from fortGLP
+    /// @dev _configData:
+    /// @dev _index - index of the vault in the vaultFactory, determines the asset and the strike vault
+    /// @dev _amount - amount of the assetVaultPrimaryAsset to deposit, will be set to the balance if set to type(uint256).max
+    /// @dev _id - id of the vault in the vaultFactory, determines end date of the Epoch
+    /// @dev _type - true for Risk Vault, false for Hedge Vault
+    function terminate(bytes memory _configData) external override onlyManager returns (uint256) {
```

Comment:
Do I understand it correctly that `execute` would be called before the Y2K epoch starts and `terminate` would be called after the Y2K epoch ended?

Reply:
Yeah I think so, once I implement tests we'll see it in practice.